packages/backend/src/apis/quadlet-api-impl.ts

```diff
@@ -57,9 +62,12 @@ export class QuadletApiImpl extends QuadletApi {
   override async stop(connection: ProviderContainerConnectionIdentifierInfo, id: string): Promise<boolean> {
     // ensure the quadlet exists & have an associated systemd service
     const quadlet = this.dependencies.quadlet.getQuadlet(id);
-    if (!quadlet.service)
+    if (!isServiceQuadlet(quadlet))
       throw new Error(`cannot stop quadlet: quadlet with id ${id} does not have an associated systemd service`);
 
+    if (isTemplateQuadlet(quadlet) && !quadlet.defaultInstance)
```

Comment:
This is the same logic in start / stop / create that could be refactored as proposed in 

https://github.com/axel7083/extension-podman-quadlet/pull/4/files

(when only two actions existed, let me know if you want me to update the PR)

Reply:
I would rather have this in a separate PR for now, to have this fix/feature merge first